camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EffectorsYamlTest.java

```diff
@@ -18,45 +18,168 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.apache.brooklyn.core.entity.EntityPredicates.displayNameEqualTo;
+import static org.testng.Assert.assertEquals;
 
 @Test
 public class EffectorsYamlTest extends AbstractYamlTest {
     private static final Logger log = LoggerFactory.getLogger(EffectorsYamlTest.class);
 
     @Test
-    public void testWithAppEnricher() throws Exception {
-        Entity app = createAndStartApplication(loadYaml("test-app-with-effectors.yaml"));
-        waitForApplicationTasks(app);
-        Assert.assertEquals(app.getDisplayName(), "test-app-with-effectors");
-
-        Entities.dumpInfo(app);
-        Thread.sleep(Duration.THIRTY_SECONDS.toMilliseconds());
-
-        Entity start1 = null, start2 = null;
-        Assert.assertEquals(app.getChildren().size(), 2);
-        for (Entity child : app.getChildren()) {
-            if (child.getDisplayName().equals("start1"))
-                start1 = child;
-            if (child.getDisplayName().equals("start2"))
-                start2 = child;
+    public void testEffectorWithVoidReturnInvokedByGetConfig() throws Exception {
+
+        Entity app = createAndStartApplication(
+            "services:",
+            "- type: " + TestEntity.class.getName(),
+            "  id: entity1",
+            "  brooklyn.config:",
+            "    test.confName: $brooklyn:entity(\"entity1\").effector(\"myEffector\")"
+        );
+        TestEntity testEntity = (TestEntity)Iterables.getOnlyElement(app.getChildren());
+
+        assertCallHistory(testEntity, "start");
+
+        // invoke the effector
+        Assert.assertNull(testEntity.getConfig(TestEntity.CONF_NAME));
+
+        assertCallHistory(testEntity, "start", "myEffector");
+    }
+
+
+    @Test(enabled = false, description = "currently not possible to say '$brooklyn:entity(\"entity1\").effector:'")
+    public void testEffectorMultiLine() throws Exception {
+
+        Entity app = createAndStartApplication(
+            "services:",
+            "- type: " + TestEntity.class.getName(),
+            "  id: entity1",
+            "  brooklyn.config:",
+            "    test.confName: ",
+            "      $brooklyn:entity(\"entity1\").effector:",
+            "      - myEffector"
+        );
+        TestEntity testEntity = (TestEntity)Iterables.getOnlyElement(app.getChildren());
+
+        assertCallHistory(testEntity, "start");
+
+        Assert.assertNull(testEntity.getConfig(TestEntity.CONF_NAME));
+
+        assertCallHistory(testEntity, "start", "myEffector");
+    }
+
+    @Test
+    public void testEffectorWithReturn() throws Exception {
+
+        Entity app = createAndStartApplication(
+            "services:",
+            "- type: " + TestEntity.class.getName(),
+            "  id: entity1",
+            "  brooklyn.config:",
+            "    test.confName: ",
+            "      $brooklyn:entity(\"entity1\").effector(\"identityEffector\", \"hello\")"
```

Comment:
Worth noting that the multi-line version that took args doesn't work (due to our DSL parsing we believe). And linking to a brooklyn jira issue, once you've created that.